packages/rollup/src/__tests__/main.test.ts

```diff
@@ -159,9 +159,42 @@ describe('rollup-plugin-imagetools', () => {
     })
 
     describe('silent', () => {
-      test('false by default', () => {})
-      test('true disables all warnings', () => {})
-      test('false enables warnings', () => {})
+      for (const [message, options, shouldPrintWarning] of [
+        ['false by default', {}, true],
+        ['true disables all warnings', { silent: true }, false],
+        ['false enables all warnings', { silent: false }, true]
+      ] as const) {
+        test(message, async () => {
+          const spy = vi.fn()
+          await rollup({
+            input: join(__dirname, '__fixtures__/index.js'),
+            onwarn: spy,
+            plugins: [
+              testEntry(`
+                            import Image from "./with-metadata.png?warn"
+                            window.__IMAGE__ = Image
+                        `),
+              imagetools({
+                ...options,
+                extendTransforms() {
+                  return [
+                    (config, context) => {
+                      context.warn('A warning')
+                      return (image) => image
+                    }
+                  ]
+                }
+              })
+            ]
+          })
+
+          if (shouldPrintWarning) {
+            expect(spy.mock.calls[0][0].message).toContain('A warning')
```

Comment:
The message logged by vite includes color codes if the terminal supports it.

```ts
expect(spy).toHaveBeenCalledWith("[plugin:imagetools] A warning")
```
Only worked if I ran `npm run test | cat` so as to disable colors. No color codes are inserted inside the warning text, so we can find it with `expect(...).toContain`.